app/src/cambodia/java/org/eyeseetea/malariacare/strategies/SettingsActivityStrategy.java

```diff
@@ -2,80 +2,53 @@
 import org.eyeseetea.malariacare.SettingsActivity;
-import org.eyeseetea.malariacare.data.sync.exporter.PushController;
-import org.eyeseetea.malariacare.layout.listeners.LoginRequiredOnPreferenceClickListener;
+import org.eyeseetea.malariacare.data.authentication.AuthenticationManager;
+import org.eyeseetea.malariacare.domain.usecase.LogoutUseCase;
 import org.eyeseetea.malariacare.layout.listeners.PullRequiredOnPreferenceChangeListener;
-import org.hisp.dhis.android.sdk.job.NetworkJob;
-import org.hisp.dhis.android.sdk.persistence.Dhis2Application;
-import org.hisp.dhis.android.sdk.persistence.preferences.ResourceType;
 
 public class SettingsActivityStrategy extends ASettingsActivityStrategy {
 
-    private static final String TAG = ".SettingsActivityStrategy";
     private PullRequiredOnPreferenceChangeListener pullRequiredOnPreferenceChangeListener;
-    private LoginRequiredOnPreferenceClickListener loginRequiredOnPreferenceClickListener;
+
+
+    private static final String TAG = ".SettingsStrategy";
+    LogoutAndLoginRequiredOnPreferenceClickListener loginRequiredOnPreferenceClickListener;
 
     public SettingsActivityStrategy(SettingsActivity settingsActivity) {
         super(settingsActivity);
 
-        loginRequiredOnPreferenceClickListener = new LoginRequiredOnPreferenceClickListener(
-                settingsActivity);
+        loginRequiredOnPreferenceClickListener =
+                new LogoutAndLoginRequiredOnPreferenceClickListener(
+                        settingsActivity);
 
         pullRequiredOnPreferenceChangeListener = new PullRequiredOnPreferenceChangeListener();
-
     }
 
     @Override
     public void onCreate() {
-        Dhis2Application.bus.register(this);
     }
 
     @Override
-    public void setupPreferencesScreen(PreferenceScreen preferenceScreen) {
+    public void onStop() {
 
     }
 
     @Override
-    public void onStop() {
-        try {
-            //Unregister from bus before leaving
-            Dhis2Application.bus.unregister(this);
-        } catch (Exception e) {
-        }
-    }
-
-    @Subscribe
-    public void callbackLoginPrePull(NetworkJob.NetworkJobResult<ResourceType> result) {
-        if (PushController.getInstance().isPushInProgress()) {
-            return;
-        }
-        //Nothing to check
-        if (result == null || result.getResourceType() == null || !result.getResourceType().equals(
-                ResourceType.USERS)) {
-            return;
-        }
-
-        //Login failed
-        if (result.getResponseHolder().getApiException() != null) {
-            new AlertDialog.Builder(settingsActivity)
-                    .setTitle(R.string.dhis_url_error)
-                    .setMessage(R.string.dhis_url_error_bad_credentials)
-                    .setNeutralButton(android.R.string.yes, null)
-                    .create()
-                    .show();
+    public void setupPreferencesScreen(PreferenceScreen preferenceScreen) {
```

Comment:
@idelcano I think that OrgUnit Preference should be remove only for myanmar but not for Lao and Cambodia